linux_os/guide/system/software/integrity/software-integrity/aide/aide_build_database/ansible/shared.yml

```diff
@@ -5,14 +5,20 @@
 # disruption = low
 {{% if 'sle' in product %}}
 {{%- set aide_init_command = "/usr/bin/aide --init" -%}}
+{{%- set packager_refresh_cmd = "zypper -q --no-remote ref" -%}}
 {{%- set aide_stage_src = "/var/lib/aide/aide.db.new" -%}}
 {{%- set aide_stage_dest = "/var/lib/aide/aide.db" -%}}
 {{% else %}}
 {{%- set aide_init_command = "/usr/sbin/aide --init" -%}}
+{{%- set packager_refresh_cmd = "echo" -%}}
 {{%- set aide_stage_src = "/var/lib/aide/aide.db.new.gz" -%}}
 {{%- set aide_stage_dest = "/var/lib/aide/aide.db.gz" -%}}
 {{% endif %}}
 
+- name: "{{{ rule_title }}} - Ensure Repositories Are Updated"
+  ansible.builtin.command: {{{ packager_refresh_cmd }}}
+  ignore_errors: True
```

Comment:
I don't like this solution. On systems other than SLE this will generate a task "Ensure repositories are updated" that contains an "echo" command. The people that will see this code will be confused. Instead, I would prefer to wrap this entire block into an `{{% if 'sle' in product %}}` Jinja block.

Reply:
Hello @jan-cerny

I was also thinking to as you, but ...I had an error in our test environment and the main reason of the error was that fact that repositories i.e. the sources of installation packages was not updated.  After the update/refresh everything was fine. Because I am not familiar with rhel*/ubuntu - do you think that similar action will be necessary for them.  For example I think that rpm --justdb works in the similar manner ( https://jfearn.fedorapeople.org/en-US/RPM/4/html/RPM_Guide/ch-command-reference.html).

Do you believe that the rules which require installation of a package always will install the package without any problem.  I am not covering the case when for some reason you don't have access to the internet, but when the execution of another rules breaks your system. For example  after the execution of an ansible remediation on SUSE  OS - of the following rules: rpm_verify_hashes and rpm_verify_permissions  - the DNS on my test machine was broken, and after for me was not possible to install nothing when the rule requires that. These rules are created for good reason to guarantee the consistence of the OS, but at the same time, they can break some of existing configurations (because for them these configurations are not correct) and to put your OS in not working state. 

Have a nice day
Rumen